django/db/models/sql/compiler.py

```diff
@@ -223,7 +223,12 @@ def get_select_from_parent(klass_info):
 
         ret = []
         for col, alias in select:
-            ret.append((col, self.compile(col, select_format=True), alias))
+            try:
+                sql, params = self.compile(col, select_format=True)
+            except EmptyResultSet:
+                # Selecting a predicate that is always false:
+                sql, params = '0', ()
```

Comment:
Case expressions use Q internally which led me to writing the above tests. I think you're right that EmptyResultSet might be able to be caught directly, and then it could use the `default` argument in place of the static `'0'` you have above. The changes need to be made in tandem though.